build.gradle

```diff
@@ -318,9 +319,8 @@ project('spring-integration-core') {
 		}
 		compile("io.fastjson:boon:$boonVersion", optional)
 		compile("com.esotericsoftware:kryo-shaded:$kryoShadedVersion", optional)
-		compile "io.micrometer:micrometer-core:$micrometerVersion"
+		compile("io.micrometer:micrometer-core:$micrometerVersion", optional)
```

Comment:
No, I think it's not going to work.
Look here: `AbstractMessageSource implements MessageSourceMetrics` and the last one has hard dependency on the `io.micrometer.core.instrument.Counter`. Therefore no any application is going to work without `micrometer-core` on CP.

Therefore this is a breaking change.

I'm OK to live with a hard dependency here. In fact it comes transitively.